### PR TITLE
client/web: populate device details view

### PR DESCRIPTION
**Pull request by soniaappasamy**
Fills /details page with real values, passed back from the /data endpoint.

Updates tailscale/corp#14335

### CODE REVIEW

File: client/web/web.go [hunk 1]
```diff
@@ -490,21 +490,35 @@ func (s *Server) serveAPI(w http.ResponseWriter, r *http.Request) {
 }
 
 type nodeData struct {
```
Comment (soniaappasamy) [inline client/web/web.go:492]: Didn't change the names of any fields. Purely additive with moving some of the fields around to a more-sensical grouping. Ideally we would add `json:` specs to each of them, and have them camelcase in the json conversion, but then we'll have to maintain two different structs until the legacy client goes away. Opted to not touch that at all in this change, we can do that in a follow-up if desired.

Comment (willnorris) [reply client/web/web.go:492]: agreed that I wouldn't worry too much  about this, given that legacy will be going away soon

Comment (soniaappasamy) [reply client/web/web.go:492]: yeah i guess we could always migrate it to camel case on the next release after legacy goes away (or the same one)

Comment (willnorris) [reply client/web/web.go:492]: will we even still have this endpoint after legacy goes away, or does it get replaced with a proxied localapi call?  🤷🏻

Comment (willnorris) [reply client/web/web.go:492]: I suppose we'll still need it, since there is additional data we're adding here that doesn't exist in localapi

Comment (soniaappasamy) [reply client/web/web.go:492]: oh yeah good point

Comment (soniaappasamy) [reply client/web/web.go:492]: hahah localapi is not `json:` specced either 😅 so maybe better to just use PascalCase here for consistency

**Review by willnorris — APPROVED** [review]

**Pull request merged**